index.js

```diff
@@ -720,8 +1224,15 @@
       });
     };
 
+    //  valueTypes :: [a] -> [Type]
+    var valueTypes = function(values) {
+      return chain(commonTypes(map(values, determineActualTypes)),
+                   rejectInconsistent);
+    };
+
```

Comment:
Thank you for defining this function! We use this expression in quite a few places.